### PR TITLE
Azure : new ethtool test case

### PR DESCRIPTION
**Pull request by asofron**
Azure
```
[LISAv2 Test Results Summary]
Test Run On           : 03/20/2019 08:37:05
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:2

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.59 

[LISAv2 Test Results Summary]
Test Run On           : 03/20/2019 08:43:46
ARM Image Under Test  : SUSE : SLES : 15 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:3

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.45 

[LISAv2 Test Results Summary]
Test Run On           : 03/20/2019 08:48:02
ARM Image Under Test  : OpenLogic : CentOS : 7.5 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:3

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.52 
```

Hyper-V 
```
[LISAv2 Test Results Summary]
Test Run On           : 03/20/2019 09:01:24
VHD Under Test        : ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:2

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.48 



[LISAv2 Test Results Summary]
Test Run On           : 03/20/2019 09:03:26
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:2

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.43 


[LISAv2 Test Results Summary]
Test Run On           : 03/20/2019 09:08:29
VHD Under Test        : sles_15.vhdx
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              ETHTOOL-GET-STATISTICS                                                            PASS                 0.41 
```

### CODE REVIEW

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#############################################################################
+ChangeMTU(){
```
Comment (chvalean) [inline Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: ()[space]{
and fix indentation inside function

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    test_iface=$1
+    value=$2
+LogMsg "Setting MTU $value on $test_iface" >> ~/summary.log
+sudo ip link set dev $test_iface mtu $value
+if [ $? -ne 0 ]; then
```
Comment (chvalean) [inline Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: use exit code function?

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+# Changing MTU
+ChangeMTU $net_interface 1505
+ChangeMTU $net_interface 2048
+ChangeMTU $net_interface 4096
+
+# Get NIC statistics using ethtool
+LogMsg "Getting NIC statistics with ethtool" >> ~/summary.log
+stats=$(ethtool -S $net_interface)
+if [ $? -ne 0 ]; then
+    LogErr "Failed get NIC statistics with ethtool !" >> ~/summary.log
```
Comment (chvalean) [inline Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: Failed to get

File: XML/TestCases/FunctionalTests-Network.xml [hunk 4]
```diff
@@ -647,4 +647,15 @@
...
+    <test>
+        <testName>ETHTOOL-GET-STATISTICS</testName>
+        <testScript>ETHTOOL-GET-STATISTICS.sh</testScript>
+        <files>.\TestScripts\Linux\ETHTOOL-GET-STATISTICS.sh,.\TestScripts\Linux\utils.sh</files>
+        <setupType>OneVM</setupType>
+        <Platform>Azure,HyperV</Platform>
+        <Category>Functional</Category>
+        <Area>NETWORK</Area>
+        <Tags>ethtool</Tags>
+        <Priority>1</Priority>
```
Comment (chvalean) [inline XML/TestCases/FunctionalTests-Network.xml]: let's make this P0

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Check if ethtool exist and install it if not
+VerifyIsEthtool
+
+net_interface=eth0
+# Verify the new NIC received an IP v4 address
+LogMsg "Verify the new NIC has an IPv4 address"
+ip addr show ${net_interface} | grep "inet\b" > /dev/null
```
Comment (chvalean) [inline Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: these 4 lines are not needed, automation will not get here if interface is not up / does not have IP.
remove lines 31-35.

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+stats=$(ethtool -S $net_interface)
+if [ $? -ne 0 ]; then
+    LogErr "Failed get NIC statistics with ethtool !" >> ~/summary.log
+    SetTestStateFailed
+    exit 0
+else
+    LogMsg "$stats" >> ~/summary.log
+fi
+LogMsg "Getting NIC statistics per CPU with ethtool" >> ~/summary.log
+statspcup=$(ethtool -S eth0 | grep 'queue_')
```
Comment (chvalean) [inline Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: use $net_interface

**Review by chvalean — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 1 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: Fixed

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 3 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: Updated

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 6 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: Fixed

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 2 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: fixed

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 5 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: fixd

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 3 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: Fixed

File: Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh (hunk 6 above)
Comment (asofron) [reply Testscripts/Linux/ETHTOOL-GET-STATISTICS.sh]: fixed

File: XML/TestCases/FunctionalTests-Network.xml (hunk 4 above)
Comment (asofron) [reply XML/TestCases/FunctionalTests-Network.xml]: solved

**Review by chvalean — APPROVED** [review]

**Pull request merged**